### PR TITLE
Functional components (apps, react-components)

### PR DESCRIPTION
**Pull request by kwingram25**
Continued

### CODE REVIEW

File: packages/apps/src/SideBar/NodeInfo.tsx [hunk 1]
```diff
@@ -15,34 +15,30 @@ interface Props extends ApiProps, BareProps {}
...
+    </div>
+  );
+}
 
-    if (!isApiReady) {
-      return null;
-    }
+function NodeInfo (props: Props): React.ReactElement<Props> {
+  const { api, className } = props;
+  const uiInfo = `apps v${pkgJson.version}`;
```
Comment (jacogr) [inline packages/apps/src/SideBar/NodeInfo.tsx]: Can actually move this out to file-level

File: packages/apps/src/SideBar/NodeInfo.tsx [hunk 2]
```diff
@@ -15,34 +15,30 @@ interface Props extends ApiProps, BareProps {}
...
+      <NodeName />&nbsp;
+      <NodeVersion label='v' />
+    </div>
+  );
+}
 
-    if (!isApiReady) {
-      return null;
-    }
+function NodeInfo (props: Props): React.ReactElement<Props> {
```
Comment (jacogr) [inline packages/apps/src/SideBar/NodeInfo.tsx]: Why not ({ Api, className }: props)

File: packages/apps/src/overlays/Connecting.tsx [hunk 3]
```diff
@@ -17,59 +17,57 @@ type Props = I18nProps & ApiProps;
...
+    return null;
   }
 
-  private renderConnecting (): React.ReactNode {
-    const { className, isApiConnected, t } = this.props;
-
-    if (isApiConnected) {
-      return null;
-    }
+  const wsUrl = settings.apiUrl;
```
Comment (jacogr) [inline packages/apps/src/overlays/Connecting.tsx]: This stuff we can keep file level. It doesn’t need to be re-evaluated.

File: packages/apps/src/overlays/Connecting.tsx [hunk 4]
```diff
@@ -17,59 +17,57 @@ type Props = I18nProps & ApiProps;
...
+  return (
+    <BaseOverlay
+      className={className}
+      icon='globe'
+    >
+      <div>{t('You are not connected to a node. Ensure that your node is running and that the Websocket endpoint is reachable.')}</div>
+      {
+        isFirefox && isWs
+          ? <div>{t('With the Firefox browser connecting to insecure WebSockets ({{wsUrl}}) will fail due to the browser not allowing localhost access from a secure site.', { replace: { wsUrl } })}</div>
+          : undefined
```
Comment (jacogr) [inline packages/apps/src/overlays/Connecting.tsx]: Completely throw-away comment - I always wonder about these where we mix null and undefined. Doesn’t make a difference either way, but we are inconsistent. 

Don’t do anything, just saw it

File: packages/apps/src/SideBar/NodeInfo.tsx (hunk 2 above)
Comment (kwingram25) [reply packages/apps/src/SideBar/NodeInfo.tsx]: Because I pass props to renderNode later :)

File: packages/apps/src/SideBar/NodeInfo.tsx (hunk 1 above)
Comment (kwingram25) [reply packages/apps/src/SideBar/NodeInfo.tsx]: How do you mean?

Comment (jacogr) [reply packages/apps/src/SideBar/NodeInfo.tsx]: So uiInfo will never change. So the constant can and should be defined outside of the function, ie. File-level.

**Pull request merged**